src/day_3_1.py

```diff
@@ -0,0 +1,32 @@
+from re import finditer
+from common import prepare_day03_input_data
+
+def get_matches(line: str) -> list:
+    matches = list()
+    for match in finditer(r"mul\((?P<factor_1>[0-9]{1,3}),(?P<factor_2>[0-9]{1,3})\)", line):
+        matches.append(match)
+    
+    return matches
```

Comment:
```suggestion
def get_matches(line: str) -> list:
    while match = next(finditer(r"mul\((?P<factor_1>[0-9]{1,3}),(?P<factor_2>[0-9]{1,3})\)", line)):
        yield match
    
    return matches
```

we can turn this function into a generator one.